Kokoro setup for building gRPC Python with Bazel

ghostwriternr opened the pull request:
In preparation for #8079, new CI tests have to be added to Kokoro for testing gRPC Python with Bazel.

- python-dev has to installed so that Python.h headers are available for building Cython code.
- ~Reinstall python-pkg-resources to fix path python uses to find packages~

cc: @nathanielmanistaatgoogle  @matt-kwong  @jtattermusch

matt-kwong commented on tools/dockerfile/test/bazel/Dockerfile:
```diff
@@ -15,7 +15,7 @@
 
 # Install basic packages and Bazel dependencies.
-RUN apt-get update && apt-get install -y software-properties-common python-software-properties
+RUN apt-get update && apt-get install -y software-properties-common python-software-properties python-dev
```
Our Dockerfiles are generated from template files because we have so many of them and a lot of them share common package installations. Can you make this change to https://github.com/grpc/grpc/blob/master/templates/tools/dockerfile/test/bazel/Dockerfile.template and then run `./tools/buildgen/generate_projects.sh`? The script will update this Dockerfile for you.

matt-kwong replied:
In fact, it's probably better if we use the standard Python installation template. You can replace that whole line in the template file with `<%include file="../../python_deps.include"/>`. Also, could you move this line to be right before `RUN apt-get install --reinstall python-pkg-resources`? Keeping the Python related installations together makes the Dockerfile more organized :)

matt-kwong commented on tools/dockerfile/test/bazel/Dockerfile:
```diff
@@ -25,6 +25,7 @@ RUN apt-get update && apt-get -y install \
   make \
   openjdk-8-jdk \
   vim
+RUN apt-get install --reinstall python-pkg-resources
```
Could you add a comment explaining the reinstall? The reasoning you gave in your pull request description would be good.

ghostwriternr commented:
@matt-kwong I've made the requested changes. Rebased to include #16006 changes too.

Reinstalling `python-pkg-resources` is no longer required to fix the path python uses to find packages, so I've removed it.

grpc-testing commented:
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

grpc-testing commented:
```
[trickle] No significant performance differences
```

nathanielmanistaatgoogle commented on tools/internal_ci/linux/grpc_python_bazel_test.cfg:
```diff
@@ -0,0 +1,23 @@
+# Copyright 2017 gRPC authors.
```
Stale year because this header was copied from someplace else?

nathanielmanistaatgoogle commented on tools/internal_ci/linux/grpc_python_bazel_test_in_docker.sh:
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+# Copyright 2017 gRPC authors.
```
Stale year here too.

ghostwriternr commented:
Updated :)

grpc-testing commented:
```
[microbenchmarks] No significant performance differences
```

grpc-testing commented:
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

grpc-testing commented:
```
[trickle] No significant performance differences
```

grpc-testing commented:
```
[microbenchmarks] No significant performance differences
```

matt-kwong commented:
Interop failures:
#16004

matt-kwong commented:
@nathanielmanistaatgoogle, if this PR looks good, we can merge it.

nathanielmanistaatgoogle commented:
@matt-kwong: looks great to me; merging.
